### PR TITLE
refactor(#1989):  Remove ATTR_* from Mojos

### PR DESCRIPTION
**Pull request by volodya-lombrozo**
1. Remove ATTR_* from all Mojos
2. Rename and simplify methods 
3. Puzzle is added for `binarizesWithoutErrors` test

---

## PR-Codex overview
This PR simplifies the `ForeignTojo` class and makes it more readable.

### Detailed summary
- Renamed `eolangObject()` to `source()`.
- Removed unused imports and commented-out code.
- Changed several method names to be more descriptive.
- Removed `Tojo` references and replaced them with `ForeignTojo`.
- Added `withSecondXmir()` method to get all `ForeignTojo` objects with an XMIR2 attribute.
- Replaced `withDiscoveredAt()` with `withDiscoveredFrom()`.
- Renamed `ForeignTojos` to `ForeignTojoCollection`.
- Removed unused `FOREIGN_TOJOS` field.

> The code diff in this PR exceeds the max number of characters, so this overview may be incomplete.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by volodya-lombrozo** [comment]
@Graur Could you have a look, please?

**Review by Graur — CHANGES_REQUESTED** [review]

**Review by Graur — APPROVED** [review]
@volodya-lombrozo Looks good to me
@yegor256 Please, have a look

File: eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java [hunk 1]
```diff
@@ -44,8 +45,13 @@ final class BinarizeMojoTest {
      * BinarizeMojo can binarize without errors.
      * @param temp Temporary directory.
      * @throws Exception If fails.
+     * @todo #1989:30min Enable BinarizeMojoTest.binarizesWithoutErrors() test. The test is disabled
+     *  because it fails on systems that don't hava install rust. We have to invent a way to check
+     *  installed rust on the system before running that test. The logic is similar to
+     *  {@link org.eolang.maven.OnlineCondition}.
```
[the remark this answers is not in the document]
Comment (volodya-lombrozo) [reply eo-maven-plugin/src/test/java/org/eolang/maven/BinarizeMojoTest.java:51]: @Graur I suggest ignoring tests for Rust only if it is not installed on the working system.

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Review by Graur — APPROVED** [review]

**Comment by volodya-lombrozo** [comment]
@yegor256 Could you merge that changes, please?

**Comment by volodya-lombrozo** [comment]
@yegor256 Reminder

**Comment by volodya-lombrozo** [comment]
@yegor256 Reminder

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**